doc/ceph-volume/lvm/zap.rst

```diff
@@ -15,18 +15,51 @@ on the given lv or partition will be removed and all data will be purged.
 
 Zapping a logical volume::
 
-      ceph-volume lvm zap {vg name/lv name}
+    ceph-volume lvm zap {vg name/lv name}
 
 Zapping a partition::
 
-      ceph-volume lvm zap /dev/sdc1
+    ceph-volume lvm zap /dev/sdc1
 
-If you are zapping a raw device or partition and would like any vgs or lvs created
-from that device removed use the ``--destroy`` flag. A common use case is to simply
-deploy OSDs using a whole raw device. If you do so and then wish to reuse that device for
-another OSD you must use the ``--destroy`` flag when zapping so that the vgs and lvs that
-ceph-volume created on the raw device will be removed.
+Removing Devices
+----------------
+When zapping, and looking for full removal of the device (lv, vg, or partition)
+use the ``--destroy`` flag. A common use case is to simply deploy OSDs using
+a whole raw device. If you do so and then wish to reuse that device for another
+OSD you must use the ``--destroy`` flag when zapping so that the vgs and lvs
+that ceph-volume created on the raw device will be removed.
+
+.. note:: Multiple devices can be accepted at once, to zap them all
 
 Zapping a raw device and destroying any vgs or lvs present::
 
-      ceph-volume lvm zap /dev/sdc --destroy
+    ceph-volume lvm zap /dev/sdc --destroy
+
+
+This action can be performed on partitions, and logical volumes as well::
+
+    ceph-volume lvm zap /dev/sdc1 --destroy
+    ceph-volume lvm zap osd-vg/data-lv --destroy
+
+
+Finally, multiple devices can be detected if filtering by OSD ID and/or OSD
+FSID. Either identifier can be used or both can be used at the same time. This
+is useful in situations where multiple devices associated with a specific ID
+need to be purged. When using the FSID, the filtering is stricter, and might
+not match other (possibly invalid) devices associated to an ID.
+
+By ID only::
+
+    ceph-volume lvm zap --destroy --osd-id 1
+
+By FSID::
+
+    ceph-volume lvm zap --destroy --osd-fsid 2E8FBE58-0328-4E3B-BFB7-3CACE4E9A6CE
+
+By both::
```

Comment:
Although, it's stricter, it's redundant.

Reply:
How is it redundant to use both?

Reply:
Assuming the setup is containerized with multiple clusters have the same OSD IDs then OSD FSID implies OSD ID so the call is redundant.

Reply:
This isn't redundant for non-containers. It is fine to use --osd-fsid in containers always.

Reply:
I just don't see the benefit of having 2 calls doing the same thing where one is weaker (OSD ID) but presumably easier than the other (osd fsid). Using OSD FSID only solves all the potential issues. Perhaps, we need to make that FSID more  easily discoverable.

Reply:
I understand that if specifying --osd-fsid only would not necessarily require passing the --osd-id. Having an exclude is doable with some argparse configuration, but I don't think it causes any issues here, it is almost like a noop in the handling code.

Reply:
@leseb would you mind cutting a new BZ that captures your ideas about --osd-fsid and why --osd-id might not work correctly for containers?  

[The BZ that this PR addresses](https://bugzilla.redhat.com/show_bug.cgi?id=1644847) which you created, stated clearly that: *The idea would be to do "ceph-volume lvm zap 0", where 0 is the OSD ID.*

This is exactly what this PR has done in addition to using FSID which I think will still work for your use case in containers. We can address the improvements you will add on the new bz in a separate PR